crates/harp/src/envvar.rs

```diff
@@ -0,0 +1,109 @@
+//
+// envvar.rs
+//
+// Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+//
+//
+
+use crate::protect::RProtect;
+use crate::r_lang;
+use crate::r_string;
+use crate::r_symbol;
+use crate::RObject;
+
+/// Set an environment variable through `Sys.setenv()`
+///
+/// This is a safe alternative to [std::env::set_var()]. On Windows in particular,
+/// using [std::env::set_var()] after R has started up (i.e. after Ark opens the
+/// R DLL and calls `setup_Rmainloop()`) has no effect. We believe this is because
+/// [std::env::set_var()] ends up calling `SetEnvironmentVariableW()` to set the
+/// environment variable, which only affects the Windows API environment space and
+/// not the C environment space that R seems to have access to after it has started
+/// up. This matters because R's `Sys.getenv()` uses the C level `getenv()` to
+/// access environment variables, which only looks at C environment space.
+///
+/// To affect the C environment space, [std::env::set_var()] would have had to call
```

Comment:
To be clear `Sys.setenv()` uses `putenv` as well and so has the same issue. I think what we are doing is using the R thread to synchronise all writes to environment variables. Probably worth mentioning this here.